### PR TITLE
bootstrap.yml: install cxxshim and frigg headers to /usr/share

### PR DESCRIPTION
**Pull request by 64**
As discussed on Discord. This allows cxxshim and frigg to be found via `pkg-config` rather than needing to do `ln` hacks.

Some things may break if they incorrectly use frigg without declaring it as a dependency, see e.g https://github.com/managarm/managarm/pull/453.

### CODE REVIEW

**Comment by ArsenArsen** [comment]
why not /usr/include/$name instead?

**Comment by 64** [comment]
> why not /usr/include/$name instead?

cc @avdgrinten

**Comment by avdgrinten** [comment]
That's even more confusing since the path to the headers is then `/usr/include/$name/$name`.

**Comment by 64** [comment]
maybe as a compromise, we could install to `/usr/include/freestanding/$name`?

**Comment by ArsenArsen** [comment]
well that'd only be necessary for cxxshim right? Frigg doesn't conflict with anything, and hence could reside globally?

**Comment by avdgrinten** [comment]
In principle yes, but doing it for all freestanding libs has the advantage that we cannot accidentally include non-freestanding headers.

I would still prefer installing to `/usr/share` instead of `/usr/include/freestanding`. We could symlinks `/usr/include/$name` to `/usr/share/$name/include/$name` but that would only be useful for build systems that do not use pkg-config anyway.

**Comment by avdgrinten** [comment]
We should also consider adding `--wrap-mode=nofallback` to the Meson packages that do not use the "symlinked subproject" hack anymore.

File: bootstrap.yml [hunk 1]
```diff
@@ -1133,6 +1135,28 @@ packages:
           'DESTDIR': '@THIS_COLLECT_DIR@'
         quiet: true
 
+  - name: cxxshim
+    labels: [aarch64]
```
Comment (Dennisbonke) [inline bootstrap.yml:1139]: While insignificant, we can probably add risc-v as a label here too.

File: bootstrap.yml [hunk 2]
```diff
@@ -1133,6 +1135,28 @@ packages:
...
+    architecture: '@OPTION:arch@'
```
Comment (Dennisbonke) [inline bootstrap.yml:1140]: Headers only, thus arch independent? Mark as noarch in that case.

File: bootstrap.yml [hunk 3]
```diff
@@ -760,6 +757,8 @@ packages:
         - 'meson'
         - '--prefix=/usr'
         - '--libdir=lib'
+        # Install to /usr/share to avoid conflicts with standard C++ headers.
```
Comment (Dennisbonke) [inline bootstrap.yml:760]: Going to put it here, but the same comments from cxxshim apply for frigg, while we’re working on it, we can fix it, saves us fixing later.

File: bootstrap.yml [hunk 4]
```diff
@@ -1133,6 +1135,28 @@ packages:
           'DESTDIR': '@THIS_COLLECT_DIR@'
         quiet: true
 
+  - name: cxxshim
```
Comment (Dennisbonke) [inline bootstrap.yml:1138]: Missing metadata, please add it. We should add missing metadata whenever we’re updating a package. I guess the maintainer of both frigg and cxxshim would be @avdgrinten?

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
Other than the change @avdgrinten requested, looks good to me. My comments are minor and I can live without them, tho the metadata thing really should be fixed.

**Comment by 64** [comment]
Blocker: I've just noticed that `host-managarm-tools` also requires frigg headers, but there's no straightforward way to satisfy that requirement now that we don't symlink frigg like we did before.

**Comment by avdgrinten** [comment]
Can this be merged now?

**Comment by 64** [comment]
Should be good to go

**Comment by avdgrinten** [comment]
Thank you!

**Pull request merged**